src/activity_handlers.cpp

```diff
@@ -758,6 +779,26 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
         case LST_INFINITE_MAP:
             // nothing, the liquid source is infinite
             break;
+        case LST_MONSTER:
+            Creature *c = g->critter_at(source_pos);
+            source_mon = dynamic_cast<monster *>(c);
```

Comment:
These two lines are redundant, you have already set up (and checked) the content of `source_mon`. The pointer is still fine at this point and does not need to be recalculated.